llmeter/experiments.py

```diff
@@ -28,6 +29,78 @@
     _disable_tqdm = True
 
 
+@dataclass
+class SweepResult:
+    results: dict[int, Result]
+    test_name: str
+    output_path: os.PathLike | str | None = None
+
+    def plot_sweep_results(
+        self, show: bool = True, format: Literal["html", "png"] = "html"
+    ):
+        # figs = plot_sweep_results([v for _, v in self.results.items()])
+        figs = plot_sweep_results(self)
+
+        # add individual color sequence for each plot
+        c_seqs = [
+            color_sequences.Bluered,
+            color_sequences.Turbo,
+            color_sequences.Sunsetdark_r,
+            color_sequences.Blackbody,
+            color_sequences.Viridis,
+            color_sequences.Plasma,
+        ]
+
+        for i, (_, f) in enumerate(figs.items()):
+            f.update_layout(colorway=c_seqs[i % len(c_seqs)])
+
+        # output_path = (
+        #     Path(self.output_path) / self.test_name if self.output_path else None
+        # )
+        output_path = Path(self.output_path)
+        if output_path:
+            # save figure to the output path
+            output_path.parent.mkdir(parents=True, exist_ok=True)
+            for k, f in figs.items():
+                f.write_html(output_path / f"{k}.{format}")
+
+        if show:
+            [f.show() for _, f in figs.items()]
+        return figs
+
+    @classmethod
+    def load(cls, load_path: Path | str, test_name: str | None = None) -> "SweepResult":
+        """Load sweep results from a directory.
+
+        Args:
+            load_path: Directory path containing the sweep results subdirectories
+            test_name: Optional name for the test. If not provided, will use the directory name
+
+        Returns:
+            SweepResult: A SweepResult object containing the loaded results
+
+        Raises:
+            FileNotFoundError: If load_path does not exist
+            ValueError: If no results are found in the directory
+        """
+        if isinstance(load_path, str):
+            load_path = Path(load_path)
+
+        if not load_path.exists():
+            raise FileNotFoundError(f"Load path {load_path} does not exist")
```

Comment:
Is it worth having an `if not load_path` section here to give a specific error message if user tries to load with a null or "" path? Currently I see:
- `AttributeError: 'NoneType' object has no attribute 'exists'` for None
- `FileNotFoundError: [Errno 2] No such file or directory: 'outputs/responses.jsonl'` for "" (which resolves to current working dir)

(In the "Compare load tests.ipynb" example, we currently initialise the two load test paths to `None` and expect the user to fill them in when running the notebook)

Reply:
Added case